src/OrderProvider.cs

```diff
@@ -546,43 +595,119 @@ private Hashtable ExistingUsers
         }
     }
 
-    private void ProcessRow(Mapping mapping, ColumnMappingCollection columnMappings, Dictionary<string, object> row)
+    private Hashtable _existingOrders = null;
+    /// <summary>
+    /// Collection of <OrderIntegrationOrderId>, <OrderId>  key value pairs
+    /// </summary>
+    private Hashtable ExistingOrders
     {
-        if (mapping != null && mapping.DestinationTable != null && mapping.DestinationTable.Name == "EcomOrders")
+        get
         {
-            if (!string.IsNullOrEmpty(SourceColumnNameForDestinationOrderCustomerAccessUserId))
+            if (_existingOrders == null)
             {
-                object accessUserId = DBNull.Value;
-                var OrderCustomerAccessUserExternalIdMapping = columnMappings.Find(cm => string.Compare(cm.DestinationColumn.Name, OrderCustomerAccessUserExternalId, true) == 0);
-                if (OrderCustomerAccessUserExternalIdMapping != null && OrderCustomerAccessUserExternalIdMapping.SourceColumn != null)
+                _existingOrders = [];
+                SqlDataAdapter ordersDataAdapter = new("select OrderId, OrderIntegrationOrderId from EcomOrders ", Connection);
```

Comment:
add WHERE clause for filtering out the null or empty IntegrationOrderIds that are not needed, also maybe you can GroupBy IntegrationOrderId so you don't then need to check if _existingOrders.ContainsKey

Reply:
@DWDBE I am not sure if GROUP BY in SQL is good enough, as it needs both SELECT columns in it for it to work
![image](https://github.com/user-attachments/assets/3be1a5d5-7c15-4b03-b5c3-57ade6451ab8)
and then it still is able to return multiple "same" values 
![image](https://github.com/user-attachments/assets/4eca6952-3336-4c5e-9a52-a594bd5b46cb)

Reply:
I could change the "Hashtable _existingOrders" to a Dictionary<string, string> and do a TryAdd() instead?

Reply:
you can try with just one column in grouping: "GroupBy OrderIntegrationOrderId"
yes, try with Dictionary<string, string>

Reply:
@DWDBE I've changed the Hashtables to Dictionary<string, string>, but can not do the GROUP BY of just one of the selected columns (see picture above, as SQL Manager fails)